### ISSUE TITLE
Automatically create dependencies for compiling LFRic infrastructure

### ISSUE THREAD

**Issue by hiker** [stfc/PSyclone#1842]
ATM the dependencies for the included LFRic infrastructure files is hardcoded. Once we update to use the latest fparser, we can automatically create this file using `$(FPARSER_ROOT)/example/create_dependencies.py 
```
cd .../infrastructure
create_dependencies.py */*.f90 */*.F90 > dependency
```

**Comment by arporter** [comment]
Probably we can implement this now?

**Comment by hiker** [comment]
Yes, I'd think so - the only (new) issue is that we have now pre-processed files. I **think** that means we can restrict to only handle .f90 files (not .F90).

**Comment by hiker** [comment]
When we switched to use lfric_core, this was added.
